tests/functional/backbeat/listDeleteMarker.js

```diff
@@ -0,0 +1,208 @@
+                }
+                expectedVersionId = data.VersionId;
+                return next();
+            }),
+        ], done);
+    });
+
+    after(done => async.series([
+        next => removeAllVersions({ Bucket: testBucket }, next),
+        next => s3.deleteBucket({ Bucket: testBucket }, next),
+    ], done));
+
+    it('should return the current version', done => {
+        makeBackbeatRequest({
+            method: 'GET',
+            bucket: testBucket,
+            queryObj: { 'list-type': 'current' },
+            authCredentials: credentials,
+        }, (err, response) => {
+            assert.ifError(err);
+            assert.strictEqual(response.statusCode, 200);
+            const data = JSON.parse(response.body);
+
+            assert.strictEqual(data.IsTruncated, false);
+            assert(!data.NextKeyMarker);
+            assert.strictEqual(data.MaxKeys, 1000);
+            assert.strictEqual(data.Contents.length, 1);
+            const key = data.Contents[0];
+            assert.strictEqual(key.Key, keyName);
+            assert.strictEqual(key.VersionId, expectedVersionId);
+            return done();
+        });
+    });
+
+    it('should return the non-current delete marker', done => {
+        makeBackbeatRequest({
+            method: 'GET',
+            bucket: testBucket,
+            queryObj: { 'list-type': 'noncurrent' },
+            authCredentials: credentials,
+        }, (err, response) => {
+            assert.ifError(err);
+            assert.strictEqual(response.statusCode, 200);
+            const data = JSON.parse(response.body);
+
+            assert.strictEqual(data.IsTruncated, false);
+            assert(!data.NextKeyMarker);
+            assert.strictEqual(data.MaxKeys, 1000);
+            assert.strictEqual(data.Contents.length, 1);
+            const key = data.Contents[0];
+            assert.strictEqual(key.Key, keyName);
+            assert.strictEqual(key.VersionId, expectedDMVersionId);
+            return done();
+        });
+    });
+
+    it('should return no orphan delete marker', done => {
+        makeBackbeatRequest({
+            method: 'GET',
+            bucket: testBucket,
+            queryObj: { 'list-type': 'orphan' },
+            authCredentials: credentials,
+        }, (err, response) => {
+            assert.ifError(err);
+            assert.strictEqual(response.statusCode, 200);
+            const data = JSON.parse(response.body);
+
+            assert.strictEqual(data.IsTruncated, false);
+            assert(!data.NextKeyMarker);
+            assert.strictEqual(data.MaxKeys, 1000);
+            assert.strictEqual(data.Contents.length, 0);
+            return done();
+        });
+    });
+});
+
+runIfMongoV1('listLifecycle with current delete marker version', () => {
+    let bucketUtil;
+    let s3;
+    let expectedVersionId;
+    const testBucket = 'bucket-for-list-lifecycle-current-dm-tests';
+    const keyName = 'key0';
+
+    before(done => {
```

Comment:
small nitpick :) may be `before` and `after` sections can be reused (convert into a generic fn perhaps) for both these test groups with an option to `delete` object. This is no way a necessity, just an observation.

Reply:
the “before” functions are different. Not sure we can “generate” them